fix #1192 apidocs for Data.common (ac.data and ac.pageData)

isao opened the pull request:
This adds two items to the class nav in the mojito api docs, and references Model.Vanilla. Technically it's `Y.Model` on the server, but it is api-compatible.
- Data.common.pageData
- Data.common.data

Commit:
fix #1192 apidocs for Data.common (ac.data and ac.pageData)

isao commented:
/ping @zhouyaoji

caridy commented on lib/app/addons/ac/data.common.js:
```diff
@@ -33,6 +34,28 @@ YUI.add('mojito-data-addon', function (Y, NAME) {
+     * @uses Model.Vanilla
+     */
+    /**
+     * <strong>Access point:</strong> <em>ac.data.*</em>
```
`<em>ac.pageData.*</em>`?

isao replied:
ah, yes, thanks-- copy and paste error

caridy commented:
We could link to http://yuilibrary.com/yui/docs/api/classes/Model.html or we can document at least the `get` and `set`, which are the common methods for `pageData` and `data`.

isao commented:
I link to the manual, and the full api docs are pulled in from Model.Vanilla. Here's a screenshot: http://cl.ly/QQvy (get & set, as well as all the other methods, get displayed when rendered by yuidoc)

Commit:
correct copy/pasted comment... s/data/pageData/

isao commented:
follow up on listing get/set explicitly-- won't work with `@uses`

Commit:
update mojit-proxy and model-vanilla apidocs

- get/set @method for data/pageData
- change type from Object to Model.Vanilla so class docs
  are linked
- update todo comments so we don't forget to update docs
  after yui/yui3#323 lands

isao commented:
travis errors were unrelated. this PR has no code changes.

```
$ wget -nv http://selenium.googlecode.com/files/selenium-server-standalone-2.30.0.jar
No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.
The build has been terminated
```

isao commented on lib/app/autoload/mojit-proxy.client.js:
```diff
@@ -50,18 +50,62 @@ YUI.add('mojito-mojit-proxy', function(Y, NAME) {
+         * and http://developer.yahoo.com/cocktails/mojito/docs/topics/mojito_data.html#sharing-data
          * @property data
-         * @type {Object}
+         * @type {Model.Vanilla}
```
while it's clear from the code on the next non-comment line, that the actual type is Y.Model, using Model.Vanilla is api-compatible and is what's used on the server. The reason I use it is so yuidoc will hyperlink to it's api docs, which are the same.

lzhan commented:
+1
